### PR TITLE
Use `pin_memory=True`, `shuffle=True` and `num_workers=0` by default

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
This change has been suggested in https://github.com/PyTorchLightning/lightning-bolts/pull/676#discussion_r685762613.

This PR changes the default values of datamodules to:
- `pin_memory=True`
- `shuffle=True`
- `num_workers=0` (Same as the default value of `Dataloader` in PyTorch: https://pytorch.org/docs/stable/data.html)

## Before submitting
- [#676] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by akihironitta** [comment]
The failing tests are irrelevant to the change in this PR.

**Review by Borda — APPROVED** [review]
